src/fileoverview_comment_transformer.ts

```diff
@@ -0,0 +1,80 @@
+import * as ts from 'typescript';
+import * as jsdoc from './jsdoc';
+
+const FILEOVERVIEW_COMMENTS: ReadonlySet<string> =
+    new Set(['fileoverview', 'externs', 'modName', 'mods', 'pintomodule']);
+
+function skipLeadingCommentsRecursively(n: ts.SourceFile) {
+  const start = n.getFullStart();
+  function skipLeadingComments(n: ts.Node) {
+    if (n.getFullStart() > start) return;
+    ts.setEmitFlags(n, ts.EmitFlags.NoLeadingComments);
+    ts.forEachChild(n, skipLeadingComments);
+  }
+  ts.forEachChild(n, skipLeadingComments);
+}
+
+/**
+ * A transformer that ensures the emitted JS file has an \@fileoverview comment that contains a
+ * \@suppress {checkTypes} annotation by either adding or updating an existing comment.
+ */
+export function transformFileoverviewComment(context: ts.TransformationContext) {
+  return (sf: ts.SourceFile) => {
+    let comments: ts.SynthesizedComment[] = [];
+    if (sf.statements.length) comments = ts.getSyntheticTrailingComments(sf.statements[0]) || [];
+
+    let fileoverviewIdx = -1;
+    for (let i = comments.length - 1; i >= 0; i--) {
+      const parsed = jsdoc.parseContents(comments[i].text);
+      if (parsed !== null && parsed.tags.some(t => FILEOVERVIEW_COMMENTS.has(t.tagName))) {
+        fileoverviewIdx = i;
+        break;
+      }
+    }
+    // Add a @suppress {checkTypes} tag to each source file's JSDoc comment,
+    // being careful to retain existing comments and their @suppress'ions.
+    // Closure Compiler considers the *last* comment with @fileoverview (or @externs or @nocompile)
+    // that has not been attached to some other tree node to be the file overview comment, and
+    // only applies @suppress tags from it.
+    // AJD considers *any* comment mentioning @fileoverview.
+    if (fileoverviewIdx === -1) {
+      // No existing comment to merge with, just emit a new one.
+      const commentText = jsdoc.getContents([
+        {tagName: 'fileoverview', text: 'added by tsickle'},
+        {tagName: 'suppress', type: 'checkTypes', text: 'checked by tsc'},
+      ]);
+      let target = ts.createNotEmittedStatement(sf);
+      target = ts.addSyntheticTrailingComment(
+          target, ts.SyntaxKind.MultiLineCommentTrivia, commentText, true);
+      return ts.updateSourceFileNode(sf, [target, ...sf.statements]);
```

Comment:
Maybe pull this block into a function?  It can be difficult to read this code when it isn't supper clear which blocks return early and which are just doing some logic.

`return addInitialFileOverviewComment(sf);`

Reply:
done.